### PR TITLE
[RCIAM-97] Create COmanageDbClient library

### PR DESCRIPTION
**Pull request by nikosev**
The `attrauthcomanage:COmanageDbClient` authentication processing filter is implemented as a COmanage Registry SQL client. This module makes queries to the database of COmanage Registry to retrieve information about the user and then process and release them to the SPs.

### CODE REVIEW

File: lib/Auth/Process/COmanageDbClient.php [hunk 1]
```diff
@@ -0,0 +1,480 @@
+<?php
+
+/**
+ * COmanage authproc filter.
+ *
+ * This class is the authproc filter to get attributes from rest api
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php]: Update description

File: lib/Auth/Process/COmanageDbClient.php [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+ *
+ *    authproc.aa = array(
+ *       ...
+ *       '60' => array(
+ *            'class' => 'attrauthcomanage:COmanageDbClient',
+ *            'coId' => 2,
+ *            'userIdAttribute' => 'eduPersonUniqueId',
+ *       ),
+ *
+ * @author Nicolas Liampotis <nliam@grnet.gr>
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php]: Update author list

File: lib/Auth/Process/COmanageDbClient.php [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+        . ' ON person.id = ident.co_person_id'
+        . ' WHERE'
+        . ' NOT name.deleted'
+        . ' AND name.name_id IS NULL'
+        . ' AND NOT mail.deleted'
+        . ' AND mail.email_address_id IS NULL'
+        . ' AND NOT link.deleted'
+        . ' AND NOT ident.deleted'
+        . ' AND ident.type = \'uid\''
+        . ' AND ident.identifier_id IS NULL'
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php]: Add checks for affiliation

**Comment by NicolasLiampotis** [comment]
- Update PR description
- Add CHANGELOG.md

File: lib/Auth/Process/COmanageDbClient.php (hunk 3 above)
Comment (nikosev) [reply lib/Auth/Process/COmanageDbClient.php]: Changes made in "Query improvements" commit

File: lib/Auth/Process/COmanageDbClient.php (hunk 1 above)
Comment (nikosev) [reply lib/Auth/Process/COmanageDbClient.php]: Changes made in "Update COmanageDbClient description and authors" commit

File: lib/Auth/Process/COmanageDbClient.php (hunk 2 above)
Comment (nikosev) [reply lib/Auth/Process/COmanageDbClient.php]: Changes made in "Update COmanageDbClient description and authors" commit

**Comment by nikosev** [comment]
- Pull request is linked to a ticket
- Added CHANGELOG.md
- Updated README.md with instructions about the new library

File: lib/Auth/Process/COmanageDbClient.php [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+    private function _showException($e)
+    {
+        $globalConfig = SimpleSAML_Configuration::getInstance();
+        $t = new SimpleSAML_XHTML_Template($globalConfig, 'attrauthcomanage:exception.tpl.php');
+        $t->data['e'] = $e->getMessage();
+        $t->show();
+        exit();
+    }
+
+    private function _targetCookie($cookieName, $payload) {
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php]: Remove method

**Comment by nikosev** [comment]
Hi @NicolasLiampotis,

I have applied the requested changes.

BR

**Pull request merged**